python/src/com/jetbrains/python/psi/types/PyTupleType.java

```diff
@@ -124,4 +126,15 @@ public int hashCode() {
     result = 31 * result + (myElementTypes != null ? Arrays.hashCode(myElementTypes) : 0);
     return result;
   }
+
+  @NotNull
+  @Override
+  public List<PyType> getElementTypes(@NotNull TypeEvalContext context) {
+    if (myHomogeneous) {
+      return Collections.singletonList(myElementTypes[0]);
+    }
+    else {
+      return Collections.singletonList(PyUnionType.union(Arrays.asList(myElementTypes)));
+    }
+  }
```

Comment:
PyTupleType has a bit another meaning of `getElementTypes` method: `getElementTypes().get(i)` should return type of `i`-th element in tuple